frontend/apps/crates/entry/jig/edit/src/edit/sidebar/state.rs

```diff
@@ -12,12 +13,55 @@ use chrono::{DateTime, Utc};
 
 use super::super::state::State as JigEditState;
 
+/// A wrapper struct for LiteModule so that the fields can be used as signals. State::modules is a
+/// MutableVec, but replacing the entire LiteModule whenever is_complete changes means that the
+/// entire item in the sidebar will be rerendered.
+///
+/// Should be constructed from a LiteModule.
+pub struct Module {
+    inner: Rc<LiteModule>,
+
+    /// Whether this module is completed.
+    pub is_complete: Mutable<bool>,
+}
```

Comment:
This was necessary so that we could send completion status's around without forcing the module in the sidebar to rerender.